### ISSUE TITLE
Currency switcher customer notice doesn't appear 

### ISSUE THREAD

**Issue by LevinMedia** [Automattic/woocommerce-payments#2941]
When testing, I've noticed that the automatic currency switcher notice doesn't appear. In order to reproduce:

Set your test site using a location and default currency (in WooCommerce Payments and in your general settings) to a location and currency that is different then where you are working from. 
In multi-currency, add your local currency to the store. 
Open a new incognito browser and navigate to your store.
You should see the store notification banner displaying the automatic currency switcher notice. 

I am observing that it updates the currency to my local currency, I'm just not getting the notification about the switch.

**Comment by LevinMedia** [comment]
additional context: p1631896155144900-slack-C01B8KNUYSW

**Comment by tpaksu** [comment]
One thing to check, if you dismissed the notice once, it won't appear again, until you delete the cookie starting with `store_notice` and following with a hash in your browser's cookies.

**Comment by oaratovskyi** [comment]
Can't reproduce so far. Will take a second look after PR reviews.

**Comment by oaratovskyi** [comment]
It shouldn't be working on localhost, but it worked on my end, so need to figure out why

**Comment by oaratovskyi** [comment]
@LevinMedia I've been testing it on different themes and asked @tpaksu and @dmallory42 for help.

Three of us could not reproduce the bug you've described but we discovered the next behavior:
The banner shows only once: that means if the user refreshes the page/opens a new tab it disappears.
Another thing: it does not show up on the `/shop` page even the first time.

What if we show the banner on every page until it's explicitly dismissed by the user?
I believe it will improve user experience if the user accidentally refreshed the page or navigated quickly from the main page. And most likely will solve this issue, since you'll keep noticing the banner until you click Dismiss or `Use ... instead`.

**Comment by LevinMedia** [comment]
@oaratovskyi 

>What if we show the banner on every page until it's explicitly dismissed by the user?

I think that sounds like a decent approach here. Let's start with that for this issue.  

Having thought about it a bit more though, I think it would be good to actually allow people to control automatically switching customers to the local currency, and opting to notifying their customers with the banner independently. I'll create a new ticket and add it to the backlog, I don't think it's something we need to do right away - I'll open an issue to keep track of the idea.

**Comment by oaratovskyi** [comment]
Can't reproduce the bug yet. Tried different themes as Storefront, Astra, Organic. Will take another look after resolving #2941.
Asked a question about steps needed to add the switcher to the theme header that would help me to reproduce the issue
p1633436319223600/1633090964.179500-slack-C01B8KNUYSW
